package.json

```diff
@@ -41,6 +41,7 @@
     "test-android-debug": "npm run test-android -- --ui",
     "coverage": "npm test -- --coverage",
     "postcoverage": "open-cli coverage/index.html",
+    "typecheck": "npm run typecheck -ws --if-present",
```

Comment:
Needing to define scripts in workspaces is a bit counter-intuitive. Why need to do this when just running `tsc -b` from the root will figure out and do incremental builds and avoid unnecessary rebuilds?

Reply:
Also not sure why the command name change

Reply:
Doesn't work right away, but feel free to try and get it to work somehow. The way the build system is setup in Ariakit seems fundamentally incompatible with how a TypeScript project is typically compiled. I'd like to try and revamp this, maybe with [tshy](https://github.com/isaacs/tshy) which I've used in many projects, but I'm looking for something minimal that unblocks the Solid work at the moment.

Reply:
Name change is just to be more specific, as this is exclusively used for type checking at moment (e.g. in CI).